qiskit/algorithms/optimizers/umda.py

```diff
@@ -0,0 +1,270 @@
+        """
+
+        gen = algorithm_globals.random.normal(
+            self.vector[0, :], self.vector[1, :], [self.size_gen, self.n_variables]
+        )
+
+        self.generation = self.generation[: int(self.elite_factor * len(self.generation))]
+        self.generation = np.vstack((self.generation, gen))
+
+    # truncate the generation at alpha percent
+    def _truncation(self):
+        """Selection of the best individuals of the actual generation.
+        Updates the generation by selecting the best individuals.
+        """
+        best_indices = self.evaluations.argsort()[: self.truncation_length]
+        self.generation = self.generation[best_indices, :]
+        self.evaluations = np.take(self.evaluations, best_indices)
+
+    # check each individual of the generation
+    def _check_generation(self, objective_function):
+        """Check the cost of each individual in the cost function implemented by the user."""
+        self.evaluations = np.apply_along_axis(objective_function, 1, self.generation)
+
+    # update the probability vector
+    def _update_vector(self):
+        """From the best individuals update the vector of normal distributions in order to the next
+        generation can sample from it. Update the vector of normal distributions
+        """
+
+        for i in range(self.n_variables):
+            self.vector[0, i], self.vector[1, i] = norm.fit(self.generation[:, i])
+            if self.vector[1, i] < self.std_bound:
+                self.vector[1, i] = self.std_bound
+
+    def minimize(
+        self,
+        fun: Callable[[POINT], float],
+        x0: POINT,
+        jac: Optional[Callable[[POINT], POINT]] = None,
+        bounds: Optional[List[Tuple[float, float]]] = None,
+    ) -> OptimizerResult:
+
+        self.history = []
+        not_better = 0
+        result = OptimizerResult()
+
+        for _ in range(self.max_iter):
+            self._check_generation(fun)
+            self._truncation()
+            self._update_vector()
+
+            best_mae_local = min(self.evaluations)
+
+            self.history.append(best_mae_local)
+            best_ind_local = np.where(self.evaluations == best_mae_local)[0][0]
+            best_ind_local = self.generation[best_ind_local]
+
+            # update the best values ever
+            if best_mae_local < self.best_mae_global:
+                self.best_mae_global = best_mae_local
+                self.best_ind_global = best_ind_local
+                not_better = 0
+
+            else:
+                not_better += 1
+                if not_better == self.dead_iter:
+                    break
+
+            self._new_generation()
+
+        result.x = self.best_ind_global
+        result.fun = self.best_mae_global
+        result.nfev = len(self.history) * self.size_gen
+
+        if self.disp:
+            print("\tNFVALS = " + str(result.nfev) + " F = " + str(result.fun))
+            print("\tX = " + str(result.x))
```

Comment:
When this only gives access to values that are in the result you could print anyway yourself I have to wonder whether its worthwhile to have. I recall some of the scipy optimizers give additional printing of internal behavior/values where that might be useful in configuring things. Eg if we kept hitting dead_iter condition would that be useful information?

Reply:
you mean to provide some information about the dead_iter variable? If so, I do not think so as this is only a counter to decide when to finish the execution.

Reply:
Okey. I think you are right. I am removing the disp attribute and all the code (getter, setter and checks) with reference to itself.